backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java

```diff
@@ -19,11 +21,64 @@ public UserService(UserDb userDb) {
         this.userDb = userDb;
     }
 
-    public Optional<List<Game>> listAllGames(String id) {
+    public List<Game> listAllGames(String id) {
         Optional<User> user = userDb.findById(id);
         if (user.isPresent()) {
-            return Optional.of(user.get().getPlayedGames());
+            return user.get().getPlayedGames();
         }
-        return Optional.empty();
+        return List.of();
     }
+
+    public List<GameSession> listAllSessions(String id) {
+        Optional<User> user = userDb.findById(id);
+        if (user.isPresent()) {
+            List<Game> gameList = user.get().getPlayedGames();
+            return gameList.stream().flatMap(game -> game.getGameSessionList().stream())
+                    .collect(Collectors.toList());
+        }
+        return List.of();
+    }
+
+
+    public List<PlayerDto> listAllPlayers(String id) {
+        Optional<User> user = userDb.findById(id);
+
+        if (user.isPresent()) {
+            List<Game> gameList = user.get().getPlayedGames();
+            List<Player> playerList = getPlayerListFromGameList(gameList);
+            List<String> nameList = getUniqueNamesFromPlayerList(playerList);
+            List<PlayerDto> playerDtoList = buildPlayerDtoList(playerList, nameList);
+            return playerDtoList;
+        }
+        return List.of();
+    }
+
+    private List<PlayerDto> buildPlayerDtoList(List<Player> playerList, List<String> nameList) {
+        List<PlayerDto> playerDtoList = nameList.stream().map(name -> {
+            Player match = playerList.stream()
+                    .filter(player -> player.getName().equals(name))
+                    .findFirst()
+                    .orElse(null);
+            return PlayerDto.builder().name(name).color(match.getColor()).build();
```

Comment:
if a user could not be found this part might cause a `NullPointerException` because you are trying to call a method (`getColor()`) on a null object
you could use `.findAny()` which should return an Optional and include an `if` statement to build an object 
```
if(match.isPresent){return PlayerDto. ... etc}
```